bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/Event.java

```diff
@@ -66,12 +69,58 @@ public boolean equals(@Nullable Object other) {
             return false;
         }
         final Event otherEvent = (Event) other;
-        return (this.title.equals(otherEvent.title) && this.start.equals(otherEvent.start)
-                && this.end.equals(otherEvent.end));
+        final String title = this.title;
+        final String comment = this.comment;
+        final String contact = this.contact;
+        final String description = this.description;
+        final String location = this.location;
+        return (this.start.equals(otherEvent.start) && this.end.equals(otherEvent.end)
+                && ((title == null && otherEvent.title == null) || (title != null && title.equals(otherEvent.title)))
```

Comment:
Such expressions can be simplified as `&& Objects.equals(this.title, otherEvent.title)`. `java.util.Objects#equals` already does the null check for us.

Reply:
Nice hint.